adyenv6b2ccheckoutaddon/acceleratoraddon/web/src/com/adyen/v6/controllers/pages/AdyenComponentController.java

```diff
@@ -75,50 +73,43 @@
-                                final HttpServletRequest request,
-                                @Valid final AdyenPaymentForm adyenPaymentForm,
-                                final BindingResult bindingResult) throws AdyenComponentException {
-        LOGGER.debug("Component paymentForm: " + adyenPaymentForm);
-
-        if(!isValidateSessionCart())
-        {
-            if (adyenPaymentForm.getBillingAddress() != null) {
-                adyenPaymentForm.resetFormExceptBillingAddress();
-            }
-            throw new AdyenComponentException("checkout.error.paymentethod.formentry.invalid");
-        }
-
-        //Save payment information
-        getAdyenCheckoutFacade().handlePaymentForm(adyenPaymentForm, bindingResult);
-        if (bindingResult.hasGlobalErrors() || bindingResult.hasErrors()) {
-            LOGGER.debug(bindingResult.getAllErrors().stream().map(DefaultMessageSourceResolvable::getCode).reduce((x, y) -> (x = x + y)));
-            if (adyenPaymentForm.getBillingAddress() != null) {
-                adyenPaymentForm.resetFormExceptBillingAddress();
-            }
-            throw new AdyenComponentException("checkout.error.paymentethod.formentry.invalid");
-        }
-
+    public String componentPayment(final HttpServletRequest request) throws AdyenComponentException {
         try {
-            validateOrderForm();
+            String requestJsonString = IOUtils.toString(request.getInputStream(), String.valueOf(StandardCharsets.UTF_8));
+            JsonObject requestJson = new JsonParser().parse(requestJsonString).getAsJsonObject();
 
-            //Make payment request
             Gson gson = new GsonBuilder().disableHtmlEscaping().create();
-            String payPalDetailsJson = adyenPaymentForm.getComponentData();
-            PayPalDetails payPalDetails = gson.fromJson(payPalDetailsJson, PayPalDetails.class);
+            Boolean termsCheck = gson.fromJson(requestJson.get("termsCheck"), Boolean.class);
+
+            if(termsCheck == null || !termsCheck) {
+                throw new InvalidCartException("checkout.error.terms.not.accepted");
+            }
+            validateOrderForm();
 
             final CartData cartData = getCheckoutFlowFacade().getCheckoutCart();
+            String paymentMethod = cartData.getAdyenPaymentMethod();
+
+            PaymentMethodDetails paymentMethodDetails;
+            if("paypal".equals(paymentMethod)) {
+                paymentMethodDetails = gson.fromJson(requestJson.get("paymentMethodDetails"), PayPalDetails.class);
+            } else if("mbway".equals(paymentMethod)) {
+                paymentMethodDetails = gson.fromJson(requestJson.get("paymentMethodDetails"), MbwayDetails.class);
+            } else {
+                throw new InvalidCartException("checkout.error.paymentethod.formentry.invalid");
+            }
+
+            cartData.setAdyenReturnUrl("https://www.adyen.com.br");//getReturnUrl(cartData.getAdyenPaymentMethod()));
```

Comment:
should this be reverted to the part in the comment?

Reply:
yes, my mistake. will fix that

Reply:
fixed